goon-universe/queries.js

```diff
@@ -1,226 +1,626 @@
+    pool.query(thisQuery,[instructorID], (error,results) => {
+        if (error) {
+            throw error
+        }
+        callback(results.rows)
+    }) 
+}
 
-    callback(`Initial Reflection updated for studentID: ${studentID}`)
-  })
+//Maybe change to addStudent and addInstructor?
+function addUser(fullName, email, callback){
+    let thisQuery= 'insert into users(full_name, email) values ($1 , $2)';
+    pool.query(thisQuery, [fullName, email], (error,results) => {
+        if (error) {
+            throw error
+        }
+        callback('Success!')
+    }) 
 }
 
-function getStakeholders(scenarioID, callback){
-  pool.query('select stakeholderid, name, designation, bio from stakeholders where scenarioid = $1', [scenarioID], (error,results) => {
-    if(error){
-      throw error
-    }
-    callback(results.rows)
-  })
+function addCourse(coursePage, courseName, semester, callback){
+    let thisQuery= 'insert into courses(webpage, name, semester) values ($1, $2, $3)';
+    pool.query(thisQuery, [coursePage, courseName, semester], (error,results) => {
+        if (error) {
+            throw error
+        }
+        callback('Success!')
+    }) 
 }
 
-function getStakeholderConvo(scenarioID, stakeholderID, callback){
-  pool.query('select conversation from stakeholders where scenarioid = $1 and stakeholderid = $2', [scenarioID, stakeholderID], (error, results) => {
-    if(error){
-      throw error
-    }
-    callback(results.rows)
-  })
+async function addReflectionResponse(studentID, input, scenarioID, timestamp, page_order) {
+	const selectPageQuery = 'select id from pages where pages.scenario_id=$1 and pages.order=$2';
+	const selectSubmissionsQuery = 'select id from submissions where submissions.scenario_id=$1 and submissions.user_id=$2';
+	const insertResponseQuery = 'INSERT INTO response(submission_id, page_num, time) VALUES ($1, $2, $3) ON CONFLICT (submission_id, page_num) DO UPDATE SET TIME = $3 RETURNING id';
+	const insertPromptResponseQuery='insert into prompt_response(id, response) VALUES ($1, $2) ON CONFLICT (id) DO UPDATE SET response = $2';
+
+	const client = await pool.connect();
+	try {
+		await client.query("BEGIN");
+		const pageSelection = await client.query(selectPageQuery, [scenarioID, page_order]);
+		let pageID = pageSelection.rows[0].id;
+		const submissionSelection = await client.query(selectSubmissionsQuery, [scenarioID, studentID]);
+		let submissionID = submissionSelection.rows[0].id;
+		// RETURNING clause returns ID at the same time
+		const responseCreation = await client.query(insertResponseQuery, [submissionID, pageID, timestamp]);
+		let responseID = responseCreation.rows[0].id;
+		await client.query(insertPromptResponseQuery, [responseID, input]);
+		await client.query("COMMIT");
+	} catch (e) {
+		await client.query("ROLLBACK");
+		throw e;
+	} finally {
+		client.release();
+	}
 }
 
-function getStakeholderHistory(scenarioID, studentID, callback){
```

Comment:
We're probably still going to need a function like this

Reply:
One of our team members is working on it on a different branch.